pkg/manifests/manifests.go

```diff
@@ -788,6 +798,32 @@ func (f *Factory) KubeStateMetricsServiceMonitor() (*monv1.ServiceMonitor, error
 	return sm, nil
 }
 
+func (f *Factory) KubeStateMetricsOperationalServiceMonitor() (*monv1.ServiceMonitor, error) {
+	sm, err := f.NewServiceMonitor(f.assets.MustNewAssetReader(KubeStateMetricsOperationalServiceMonitor))
+	if err != nil {
+		return nil, err
+	}
+
+	sm.Spec.Endpoints[0].TLSConfig.ServerName = fmt.Sprintf("kube-state-metrics.%s.svc", f.namespace)
+	sm.Spec.Endpoints[1].TLSConfig.ServerName = fmt.Sprintf("kube-state-metrics.%s.svc", f.namespace)
```

Comment:
we should be able to remove this Go code and just generate the correct manifests from jsonnet. The reason why it's done this way for now is because it was envisioned at some point that the monitoring namespaces could be customized which isn't something that will happen any time soon (if it ever happens). I'd be ok to have another PR removing all the `...ServerName = fmt.Sprintf("...", ...)` lines in pkg/manifests/manifests.go wherever it isn't required. The same goes for the `sm.Namespace = f.namespace` line below.

Reply:
I'll open a PR refactoring this and then I'll rebase the PR that lands sooner

Reply:
https://github.com/openshift/cluster-monitoring-operator/pull/1873